sdk/synapse/Azure.Analytics.Synapse.Spark/samples/Sample2_ExecuteSparkStatement.md

```diff
@@ -0,0 +1,200 @@
+
+```C# Snippet:CreateSparkStatement
+SparkStatementOptions sparkStatementRequest = new SparkStatementOptions
+{
+    Kind = SparkStatementLanguageType.Spark,
+    Code = @"print(""Hello world\n"")"
+};
+SparkStatement statementCreated = client.CreateSparkStatement(sessionCreated.Id, sparkStatementRequest);
+
+// Wait operation completion
+statementCreated = PollSparkStatement(client, sessionCreated.Id, statementCreated);
+```
+
+## Retrieve a statement
+
+To retrieve an existing statement call `GetSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:GetSparkStatement
+SparkStatement statement = client.GetSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is returned with id {statement.Id} and state {statement.State}");
+```
+
+## Cancel a statement
+
+To cancel a submitted statement call `CancelSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:CancelSparkStatement
+SparkStatementCancellationResult cancellationResult = client.CancelSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is cancelled with message {cancellationResult.Msg}");
+```
+
+## Cancel a session
+
+To cancel the entire Spark session  call `CancelSparkSession`, passing in the session ID.
+
+```C# Snippet:CancelSparkSession
+Response operation = client.CancelSparkSession(sessionCreated.Id);
+```
+
+## Support Code
+
+Today the following support code is needed to poll the status of submitted Spark statements and sessions.
+
+It is hoped to be temporary.
+
+```C# Snippet:TemporarySparkSupportCode
+ private const string Error = "error";
+ private const string Dead = "dead";
+ private const string Success = "success";
+ private const string Killed = "killed";
+ private const string Idle = "idle";
+
+public static List<string> SessionSubmissionFinalStates = new List<string>
+ {
+     Idle,
+     Error,
+     Dead,
+     Success,
+     Killed
+ };
+
+ public static SparkSession PollSparkSession(
+     SparkSessionClient client,
+     SparkSession session,
+     IList<string> livyReadyStates = null)
+ {
+     if (livyReadyStates == null)
+     {
+         livyReadyStates = SessionSubmissionFinalStates;
+     }
+
+     return Poll(
+         session,
+         s => s.Result.ToString(),
+         s => s.State,
+         s => client.GetSparkSession(s.Id, true),
+         livyReadyStates);
+ }
+
+ private const string Starting = "starting";
+ private const string Waiting = "waiting";
+ private const string Running = "running";
+ private const string Cancelling = "cancelling";
+
+ private static List<string> ExecutingStates = new List<string>
+ {
+     Starting,
+     Waiting,
+     Running,
+     Cancelling
+ };
+
+ private static SparkStatement PollSparkStatement(
+     SparkSessionClient client,
+     int sessionId,
+     SparkStatement statement)
+ {
+     return Poll(
+         statement,
+         s => null,
```

Comment:
I'm not sure that `s` offers a lot of context here.   Maybe consider a more expressive param name?